### PR TITLE
bpo-15248: Emit a compiler warning when missed a comma before tuple or list.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue15248

### CODE REVIEW

**Review by SylvainDe — COMMENTED** [review]
This looks very cool!

As an additional comment/question: would it make sense to move the logic generating the type description from the expr_ty into a function on its own to limit duplication ? I gave it a try at https://pastebin.com/XLZ2B7Cq (not tested whatsoever).

**Review by SylvainDe — COMMENTED** [review]
I like this!

File: Python/compile.c [hunk 1]
```diff
@@ -3766,6 +3766,122 @@ compiler_compare(struct compiler *c, expr_ty e)
...
+    case ListComp_kind:
+    case Dict_kind:
+    case DictComp_kind:
+    case Set_kind:
+    case SetComp_kind:
+    case GeneratorExp_kind:
+    case JoinedStr_kind:
+    case FormattedValue_kind:
+        return compiler_warn(c, "'%.200s' object is not callable, "
+                                "perhaps missed a comma?",
```
Comment (gvanrossum) [inline Python/compile.c]: “perhaps you missed a comma?”

Also use “;” instead of “,” on the previous line.

File: Python/compile.c [hunk 2]
```diff
@@ -3766,6 +3766,122 @@ compiler_compare(struct compiler *c, expr_ty e)
...
+        {
+            return 1;
+        }
+        /* fall through */
+    case Set_kind:
+    case SetComp_kind:
+    case GeneratorExp_kind:
+    case Lambda_kind:
+        return compiler_warn(c, "'%.200s' object is not subscriptable, "
+                                "perhaps missed a comma?",
```
Comment (gvanrossum) [inline Python/compile.c]: Ditto.

**Review by gvanrossum — APPROVED** [review]
Love this! Just some suggestions on the error messages.

**Comment by serhiy-storchaka** [comment]
Thank you @SylvainDe! I also thought about this type of simplification, but was hesitant. I initially implemented this idea independently, but then I seen your nice variant and borrowed it.

**Pull request merged**